static/js/unread_ops.js

```diff
@@ -15,6 +15,7 @@ function process_newly_read_message(message, options) {
         message_list.narrowed.show_message_as_read(message, options);
     }
     notifications.close_notification(message);
+    recent_topics.update_topic_unread_count(message);
```

Comment:
Just to make sure I understand this -- this is the way we trigger a rerender of the recent topics row after the new message arrives?  (And the data structure was already updated?)

Reply:
We need to update recent topics is many cases:
* New message arrives via event (also updates unread count in the same call)  - we do process_messages and rerender.
* We fetch messages in bulk manually - we do process_messages and rerender.
* A message is read -we update recent_topics by`update_topic_unread_count(message);